tests/test_frameworks.py

```diff
@@ -0,0 +1,235 @@
+        # https://github.com/numpy/numpy/issues/17640
+        X = np.float64(X)
+        y = np.int64(y)
+        y_og = np.int64(y_og)
+
+    return {
+        "X": X,
+        "y": y,
+        "y_og": y_og,
+        "error_indices": error_indices,
+        "num_classes": num_classes,
+        "num_features": num_features,
+    }
+
+
+def make_rare_label(data):
+    """Makes one label really rare in the dataset."""
+    data = deepcopy(data)
+    y = data["y"]
+    class0_inds = np.where(y == 0)[0]
+    if len(class0_inds) < 1:
+        raise ValueError("Class 0 too rare already")
+    class0_inds_remove = class0_inds[1:]
+    if len(class0_inds_remove) > 0:
+        y[class0_inds_remove] = 1
+    data["y"] = y
+    return data
+
+
+SEED = 1
+np.random.seed(SEED)
+random.seed(SEED)
+if python_version_ok():
+    tf.random.set_seed(SEED)
+    tf.keras.utils.set_random_seed(SEED)
+    torch.manual_seed(SEED)
+    torch.backends.cudnn.deterministic = True
+    torch.backends.cudnn.benchmark = False
+    torch.cuda.manual_seed_all(SEED)
+
+DATA = dataset_w_errors()
+DATA_RARE_LABEL = make_rare_label(DATA)
+
+
+@pytest.mark.skipif("not python_version_ok()", reason="need at least python 3.7")
+@pytest.mark.parametrize("batch_size,shuffle_config", [(1, 0), (32, 0), (32, 1), (32, 2)])
+def test_tensorflow(batch_size, shuffle_config, data=DATA, hidden_units=128):
+    dataset_tf = tf.data.Dataset.from_tensor_slices((data["X"], data["y"]))
+    if shuffle_config == 0:  # proper shuffling for SGD
+        dataset_shuffled = dataset_tf.shuffle(buffer_size=len(data["X"]))
+    elif shuffle_config == 1:  # shuffling for datasets that don't fit in memory
+        dataset_shuffled = dataset_tf.shuffle(buffer_size=60)
+    else:
+        dataset_shuffled = dataset_tf  # no shuffling
+
+    dataset_og_order = dataset_tf.batch(batch_size)
+    dataset_tf = dataset_shuffled.batch(batch_size)
+
+    model = KerasWrapper(
+        [
+            tf.keras.layers.Dense(
+                hidden_units, input_shape=[data["num_features"]], activation="relu"
+            ),
+            tf.keras.layers.Dense(data["num_classes"]),
+        ],
+    )
+
+    # Test base model works:
+    model.fit(
+        X=dataset_tf,
+        y=data["y"],
+        epochs=2,
+    )
+    preds_base = model.predict_proba(dataset_tf)
+
+    # Test CleanLearning performs well:
+    cl = CleanLearning(model)
+    cl.fit(dataset_tf, data["y"], clf_kwargs={"epochs": 10}, clf_final_kwargs={"epochs": 15})
+
+    preds = cl.predict(dataset_og_order)
+    err = np.sum(preds != data["y_og"]) / len(data["y_og"])
+    issue_indices = list(cl.label_issues_df[cl.label_issues_df["is_label_issue"]].index.values)
+    assert issue_indices == data["error_indices"]
+    assert err < 1e-3
+
+
+@pytest.mark.skipif("not python_version_ok()", reason="need at least python 3.7")
+@pytest.mark.parametrize("batch_size", [1, 32])
```

Comment:
You may want to test for more variations of datasets across modality, shape (length and num classes), 2d vs 3d, etc.